src/librustdoc/passes/propagate_stability.rs

```diff
@@ -39,6 +39,16 @@ impl DocFolder for StabilityPropagator<'_, '_> {
                 let item_stability = self.cx.tcx.lookup_stability(def_id);
                 let inline_stability =
                     item.inline_stmt_id.and_then(|did| self.cx.tcx.lookup_stability(did));
+                let is_glob_export = item.inline_stmt_id.and_then(|id| {
+                    let hir_id = self.cx.tcx.local_def_id_to_hir_id(id);
+                    Some(matches!(
+                        self.cx.tcx.hir_node(hir_id),
+                        rustc_hir::Node::Item(rustc_hir::Item {
+                            kind: rustc_hir::ItemKind::Use(_, rustc_hir::UseKind::Glob),
+                            ..
+                        })
+                    ))
+                });
```

Comment:
```suggestion
                let is_glob_export = item.inline_stmt_id.map(|id| {
                    let hir_id = self.cx.tcx.local_def_id_to_hir_id(id);
                    matches!(
                        self.cx.tcx.hir_node(hir_id),
                        rustc_hir::Node::Item(rustc_hir::Item {
                            kind: rustc_hir::ItemKind::Use(_, rustc_hir::UseKind::Glob),
                            ..
                        })
                    )
                });
```

Hey @tapanprakasht , sorry for the drive-by, but there is no reason to use `and_then` when your closure can never return `None`. `map` is simpler. 😊